### PR TITLE
Adding custom resource limits to k8s service

### PR DESCRIPTION
**Pull request by juandiegopalomino**
# Description

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
YOUR_ANSWER

### CODE REVIEW

File: modules/aws_k8s_service/aws-k8s-service.yaml [hunk 1]
```diff
@@ -212,6 +215,9 @@ inputs:
     description: The max amount of helm revisions to keep track of (0 for infinite)
     default: 25
 extra_validators:
+  resources:
+    cpu: any(int(min=1), regex('^[1-9][0-9]{0,3}$'), required=True)
```
Comment (loomsen) [inline modules/aws_k8s_service/aws-k8s-service.yaml:219]: I think the regex could very well be {0,4}, maybe someone wants to assign 10 or more cores for a really CPU intensive batch process or so?

File: modules/azure_k8s_service/azure-k8s-service.yaml [hunk 2]
```diff
@@ -183,6 +186,9 @@ inputs: # (what users see)
     description: The max amount of helm revisions to keep track of (0 for infinite)
     default: 25
 extra_validators:
+  resources:
+    cpu: any(int(min=1), regex('^[1-9][0-9]{0,3}$'), required=True)
```
Comment (loomsen) [inline modules/azure_k8s_service/azure-k8s-service.yaml:190]: Same as above

File: modules/gcp_k8s_service/gcp-k8s-service.yaml [hunk 3]
```diff
@@ -203,6 +206,9 @@ inputs:
     description: The max amount of helm revisions to keep track of (0 for infinite)
     default: 25
 extra_validators:
+  resources:
+    cpu: any(int(min=1), regex('^[1-9][0-9]{0,3}$'), required=True)
```
Comment (loomsen) [inline modules/gcp_k8s_service/gcp-k8s-service.yaml:210]: same as above

File: modules/local_k8s_service/local-k8s-service.yaml [hunk 4]
```diff
@@ -172,6 +175,9 @@ inputs:
     description: The max amount of helm revisions to keep track of (0 for infinite)
     default: 25
 extra_validators:
+  resources:
+    cpu: any(int(min=1), regex('^[1-9][0-9]{0,3}$'), required=True)
```
Comment (loomsen) [inline modules/local_k8s_service/local-k8s-service.yaml:179]: .

File: modules/aws_k8s_service/aws-k8s-service.yaml (hunk 1 above)
Comment (loomsen) [inline modules/aws_k8s_service/aws-k8s-service.yaml:219]: I think the regex could very well be {0,4}, maybe someone wants to assign 10 or more cores for a really CPU intensive batch process or so?

File: modules/azure_k8s_service/azure-k8s-service.md [hunk 5]
```diff
@@ -38,13 +38,28 @@ The autoscaler then uses this logic to try and balance the cpu/memory usage at t
...
+on its running node. The resource limits, however, specify the max amount of resources allocated to a pod at any time.
+These limits exist for (obvious) safety measures (e.g. preventing one app from starving the whole cluster). By default,
+this limit is set in Opta to twice the resource request value, but is exposed to users for further configuration.
+
+Please refer to the official
+[kubernetes resource management page](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)
+for more info.
+
+**NOTE**: Based on reasons outlined [here](https://github.com/robusta-dev/alert-explanations/wiki/CPUThrottlingHigh-(Prometheus-Alert)#why-you-dont-need-cpu-limits)
+Opta does not add CPU limits, only memory.
```
Comment (RemyDeWolf) [inline modules/azure_k8s_service/azure-k8s-service.md:62]: 👍

**Review by RemyDeWolf — APPROVED** [review]

**Pull request merged**